examples/errors/errors.go

```diff
@@ -0,0 +1,33 @@
+// Errors happen. This example shows how to detect and handle some of them.
+
+package main
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/anaskhan96/soup"
+)
+
+func main() {
+	_, err := soup.Get("this url isn't real!")
+	if err != nil && err.(soup.Error).Type == soup.ErrInGetRequest {
+		// Handle as required!
+	}
+
+	url := fmt.Sprintf("https://xkcd.com/50")
+	xkcd, err := soup.Get(url)
+	if err != nil {
+		// Handle it
+	}
+	xkcdSoup := soup.HTMLParse(xkcd)
+	links := xkcdSoup.Find("div", "id", "linkz")
+	if links.Error != nil && links.Error.(soup.Error).Type == soup.ErrElementNotFound {
```

Comment:
@gdholtslander not such a huge fan of the type casting here
Why can't we just have `Root.Error` be a field of our custom error type?
That way, this line can be simplified to
```go
if links.Error != nil && links.Error.Type == soup.ErrElementNotFound {
```

Reply:
We're implementing the `error` interface and overriding with our own `Error()` method, which is right, i'm just thinking if it has any benefits

Reply:
The biggest benefit is that it supports backwards compatibility. If you don't need the types you can continue exactly the way you were before by checking if the error is nil and doing your string checking if you wish (i.e. what I was doing, `strings.Contains(root.Error, "element not found")`). But once you're ready you can start using the types to check for certain errors instead.

I did make the realization while I was doing this that the actual errors we return from each method are always the same. For example, `Find` doesn't return `ElementNotFound` in some cases and `NextSiblingNotFound` in other cases. So there isn't a _ton_ of value in this PR, now that I've made this realization 🤦‍♂ But it was fun to do - and perhaps it is still useful to remove string checking from a project. And perhaps if we change how we compose some of our functions in the future, it would become more useful.

Reply:
Ah, the backward compatibility makes sense.

Yes this will certainly be useful when the library grows 👍🏻